src/components/table/columns.ts

```diff
@@ -34,6 +39,32 @@ export class ColumnDefinitionFactory {
     }
 }
 
+/**
+ * Formats the header of the column
+ *
+ * @param {Column} column the configuration for the column
+ *
+ * @returns {string | HTMLElement} custom component that renders a column header
+ */
+export const formatHeader = (column: Column) => (): string | HTMLElement => {
+    const element = document.createElement('div');
+    element.innerText = column.title;
+    if (!column.headerComponent) {
+        return;
+    }
+
+    const menu = document.createElement(column.headerComponent.name);
+    menu.setAttribute('class', 'tabulator-header-menu-button');
+
+    const props = column.headerComponent.props;
+
+    forIn(props, (value, key) => (menu[key] = value));
+
+    element.appendChild(menu);
+
+    return element;
```

Comment:
According to the interface, `headerComponent` is of the type `TableComponentDefinition`
However, `TableComponentDefinition` has a property called `propsFactory`. That is not taken care of in this function right now. Can we get that to work as well? :slightly_smiling_face:

Reply:
Is this to use for example `context`?

Reply:
Yes! :slightly_smiling_face: